pyomo/core/base/set.py

```diff
@@ -3303,7 +3344,7 @@ def __len__(self):
         )
 
     def __str__(self):
-        if self.parent_block() is not None:
+        if self._name is not None:
             return self.name
```

Comment:
This is checking for `self._name` and returning `self.name`. Is that correct? What's the difference between `_name` and `name`?

Reply:
This is intentional: `self._name` is `None` if it is an anonymous set.  If it is not None, then `_name` contains the local (block-scope) name of the Set.  `.name` is a property that returns the fully-qualified name of the Set (including all parent block names).